### PR TITLE
Fix stable/unstable Node.js detection for Node.js >=4.0.0

### LINKED ISSUE #39: Node.js follows semver; versions >1.0.0 are stable regardless of their minors
Node.js follows semver; versions below 1.0.0 technically don't have to abide to anything and the first Node with version `>=1.0.0`, i.e. `4.0.0` will not have pre-releases marked by minor numbers.

[This check](https://github.com/heroku/semver.io/blob/eaf9ba03fce55aa8e6171199bbd4397f6d8e3f77/lib/sources/node.js#L43) should be corrected.

### PR DESCRIPTION
**Pull request by mzgol**
Examples:
1. Node.js 4.1.0 is now considered stable.
2. Node.js 4.2.0-alpha.1 is now considered unstable.
3. Node.js pre-releases are now included in all but not stable,
   e.g. v0.10.16-isaacs-manual.

Note that newer semver doesn't match pre-release versions so eagerly, e.g.
"^4.2.0-alpha" will match version "4.2.0-alpha.4" but will not match
"4.2.1-alpha.4".

Fix #39

### CODE REVIEW

File: test/fixtures/node.html [hunk 1]
```diff
@@ -5,6 +5,9 @@
 <a href="latest/">latest/</a>                                            09-Jun-2014 17:45                   -
 <a href="nightlies/">nightlies/</a>                                         06-Aug-2013 22:52                   -
 <a href="npm/">npm/</a>                                               23-May-2014 16:55                   -
+<a href="v4.2.0-alpha.3/">v4.2.0-alpha.3/</a>                                           2-Mar-2016 06:25                   -
```
Comment (mzgol) [inline test/fixtures/node.html]: These versions don't exist yet but it was the only way to test them.

File: test/node/source.test.coffee [hunk 2]
```diff
@@ -34,12 +34,12 @@ describe "Node Source", ->
 
     it "has an array of all versions", ->
       assert.equal typeof(this.s.all), "object"
-      assert.equal this.s.all.length, 215
-      assert.equal this.s.all[214], '0.11.13'
+      assert.equal this.s.all.length, 219
```
Comment (mzgol) [inline test/node/source.test.coffee]: 3 new versions added (`4.0.0`, `4.1.0` & `4.2.0-alpha.3`) + `0.10.16-isaacs-manual` is now matched as well.

File: test/node/source.test.coffee [hunk 3]
```diff
@@ -34,12 +34,12 @@ describe "Node Source", ->
...
+      assert.equal this.s.all[214], '0.11.12'
 
     it "has an array of stable versions", ->
       assert.equal typeof(this.s.stable), "object"
-      assert.equal this.s.stable.length, 106
+      assert.equal this.s.stable.length, 108
```
Comment (mzgol) [inline test/node/source.test.coffee]: 2 new stable versions: `4.0.0` & `4.1.0`.

File: test/resolver.test.coffee [hunk 4]
```diff
@@ -18,17 +18,17 @@ describe "Resolver", ->
       assert.equal this.r.satisfy("0.8.15"), "0.8.15"
 
     it "matches common patterns to stable version", ->
-      assert.equal this.r.satisfy("0.10.x"), this.r.getLatestStable()
-      assert.equal this.r.satisfy("~0.10.0"), this.r.getLatestStable()
+      assert.equal this.r.satisfy("4.1.x"), this.r.getLatestStable()
+      assert.equal this.r.satisfy("~4.1.0"), this.r.getLatestStable()
```
Comment (mzgol) [inline test/resolver.test.coffee]: Matching rules for unstable versions changed in newer semver, see the commit description.

File: test/resolver.test.coffee [hunk 5]
```diff
@@ -18,17 +18,17 @@ describe "Resolver", ->
...
       assert.equal this.r.satisfy(">0.4"), this.r.getLatestStable()
       assert.equal this.r.satisfy(">=0.6.9"), this.r.getLatestStable()
       assert.equal this.r.satisfy("*"), this.r.getLatestStable()
 
     it "uses latest unstable version when request version is beyond stable version", ->
-      assert.equal this.r.satisfy("0.11.x"), this.r.getLatest()
-      assert.equal this.r.satisfy("~0.11.0"), this.r.getLatest()
-      assert.equal this.r.satisfy(">0.11.0"), this.r.getLatest()
-      assert.equal this.r.satisfy(">=0.10.100"), this.r.getLatest()
+      assert.equal this.r.satisfy("^4.2.0-alpha"), this.r.getLatest()
```
Comment (mzgol) [inline test/resolver.test.coffee]: Matching rules for unstable versions changed in newer semver, see the commit description.

**Comment by hunterloftis** [comment]
Really fantastic PR, textbook, thanks @mzgol!

**Comment by zeke** [comment]
:clap:

**Pull request merged**